cli/pcluster/configure/networking.py

```diff
@@ -0,0 +1,264 @@
+        stack_output = _create_network_stack(aws_region_name, self, parameters)
+        return {"master_subnet_id": get_stack_output_value(stack_output, "PublicSubnetId"), "use_public_ips": "true"}
+
+
+class PublicPrivateNetworkConfig(BaseNetworkConfig):
+    """The publicprivate configuration that creates one public subnet for master and one private subnet for compute."""
+
+    def __init__(self):
+        super(PublicPrivateNetworkConfig, self).__init__(
+            config_type="Master in a public subnet and compute fleet in a private subnet",
+            template_name="public-private",
+            stack_name_prefix="pubpriv"
+        )
+
+    def get_cfn_parameters(self, aws_region_name, vpc_id, internet_gateway_id, public_cidr, private_cidr):
+        """Create cloudformation-compatible stack parameter given the variables."""
+        parameters = super(PublicPrivateNetworkConfig, self)._get_cfn_parameters(
+            aws_region_name, vpc_id, internet_gateway_id
+        )
+        parameters.append({"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr})
+        parameters.append({"ParameterKey": "PrivateCIDR", "ParameterValue": private_cidr})
+        return parameters
+
+    def _create(self, aws_region_name, vpc_id, vpc_cidr, subnet_cidrs, internet_gateway_id, compute_subnet_size):  # noqa D102
+        public_cidr = evaluate_cidr(
+            vpc_cidr=vpc_cidr, occupied_cidr=subnet_cidrs, target_size=MASTER_SUBNET_IPS
+        )
+        _validate_cidr(public_cidr)
+        subnet_cidrs.append(public_cidr)
+        private_cidr = get_subnet_cidr(
+            vpc_cidr=vpc_cidr, occupied_cidr=subnet_cidrs, min_subnet_size=compute_subnet_size
+        )
+        _validate_cidr(private_cidr)
+        parameters = self.get_cfn_parameters(aws_region_name, vpc_id, internet_gateway_id, public_cidr, private_cidr)
+        stack_output = _create_network_stack(aws_region_name, self, parameters)
+        return {
+            "master_subnet_id": get_stack_output_value(stack_output, "PublicSubnetId"),
+            "compute_subnet_id": get_stack_output_value(stack_output, "PrivateSubnetId"),
+            "use_public_ips": "false",
+        }
+
+
+class NetworkConfiguration(Enum):
+    """Contain all possible network configuration."""
+
+    # py2.7 compatibility, need to specify the order
+    __order__ = "PUBLIC_PRIVATE PUBLIC"
+
+    PUBLIC_PRIVATE = PublicPrivateNetworkConfig()
+    PUBLIC = PublicNetworkConfig()
+
+
+#  FIXME update TemplateURL
+def _create_network_stack(aws_region_name, configuration, parameters):
+    LOGGER.info("Creating CloudFormation stack...")
+    LOGGER.info("Do not leave the terminal until the process has finished")
+    stack_name = "parallelclusternetworking-{0}{1}".format(configuration.stack_name_prefix, TIMESTAMP)
+    version = pkg_resources.get_distribution("aws-parallelcluster").version
+    s3_suffix = ".cn" if aws_region_name.startswith("cn") else ""
```

Comment:
We should define a function for it:

`_get_s3_suffix(aws_region)` to put in common with the other CLI code
and/or we could create a function: `get_templates_bucket`